lib/SPIRV/libSPIRV/SPIRVInstruction.h

```diff
@@ -2488,13 +2488,45 @@ class SPIRVControlBarrier : public SPIRVInstruction {
     return getValues(Operands);
   }
 
+  SPIRVCapVec getRequiredCapability() const override {
+    if (isDeviceBarrier()) {
+      return getVec(internal::CapabilityDeviceBarrierINTEL);
+    }
+    return SPIRVInstruction::getRequiredCapability();
+  }
+  std::optional<ExtensionID> getRequiredExtension() const override {
+    if (isDeviceBarrier()) {
+      return ExtensionID::SPV_INTEL_device_barrier;
+    }
+    return std::nullopt;
+  }
+
 protected:
   _SPIRV_DEF_ENCDEC3(ExecScope, MemScope, MemSema)
   void validate() const override {
     assert(OpCode == OC);
     assert(WordCount == 4);
     SPIRVInstruction::validate();
   }
+
+  bool isDeviceBarrier() const {
+    bool CanUseDeviceBarrier = getModule()->isAllowedToUseExtension(
+        ExtensionID::SPV_INTEL_device_barrier);
```

Comment:
May be to avoid breaking changes rewrite the function like this?
```suggestion
    if (!getModule()->isAllowedToUseExtension(ExtensionID::SPV_INTEL_device_barrier))
      return false;
```